### PR TITLE
add a skeleton framework for the dart2_fix app

### PR DESCRIPTION
**Pull request by devoncarew**
@scheglov @kevmoo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# dart2_fix
+# dart2_fix [![Build Status](https://travis-ci.org/dart-lang/dart2_fix.svg?branch=master)](https://travis-ci.org/dart-lang/dart2_fix)
```
Comment (kevmoo) [inline README.md]: Don't need a title at the top. In every place where it's displayed, there is sufficient context.

File: pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: dart2_fix
+description: A tool to migrate API usage to Dart 2.
+version: 1.0.0
```
Comment (kevmoo) [inline pubspec.yaml]: `1.0.0-dev` until we're ready to ship...

File: bin/dart2_fix.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:dart2_fix/src/dart2_fix.dart' as dart2_fix;
+
+main(List<String> args) async {
+  dart2_fix.ExitCode result = await dart2_fix.main(args);
```
Comment (scheglov) [inline bin/dart2_fix.dart]: Maybe name it `ExitResult` then, if the variable is `result`?

File: bin/dart2_fix.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  if (!result.isOk && result.errorMessage != null) {
+    stderr.writeln(result.errorMessage);
+  }
+  exit(result.statusCode);
```
Comment (scheglov) [inline bin/dart2_fix.dart]: `statusCode` => `exitCode`?

File: pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+environment:
+  sdk: '>=1.20.1 <2.0.0'
```
Comment (kevmoo) [inline pubspec.yaml]: Should be limited to `>=2.0.0-dev.29.0` – where the new consts were documented...

**Review by scheglov — APPROVED** [review]

**Review by kevmoo — APPROVED** [review]

File: bin/dart2_fix.dart (hunk 3 above)
Comment (devoncarew) [reply bin/dart2_fix.dart]: done

File: bin/dart2_fix.dart (hunk 4 above)
Comment (devoncarew) [reply bin/dart2_fix.dart]: done

File: README.md (hunk 1 above)
Comment (devoncarew) [reply README.md]: done

File: pubspec.yaml (hunk 2 above)
Comment (devoncarew) [reply pubspec.yaml]: done

File: pubspec.yaml (hunk 5 above)
Comment (devoncarew) [reply pubspec.yaml]: would like this more open ended in terms of what it'll install into, so that why it won't install isn't a mystery to users

**Pull request merged**